include/pros/motors.hpp

```diff
@@ -129,23 +129,41 @@ class Motor {
 	 * }
 	 * \endcode
 	 */
-	explicit Motor(const std::uint8_t port, const Motor_Gears gearset, const bool reverse,
-	               const Motor_Units encoder_units);
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const Motor_Gears gearset, const bool reverse,
+	                     const Motor_Units encoder_units);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const pros::Color gearset_color, const bool reverse,
+	                     const Motor_Units encoder_units);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const Motor_Gears gearset, const bool reverse);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const pros::Color gearset_color, const bool reverse);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const Motor_Gears gearset);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const pros::Color gearset_color);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports, const bool reverse);
+
+	explicit Motor_Group(std::initializer_list<std::int8_t> ports);
 
-	explicit Motor(const std::uint8_t port, const pros::Color gearset_color, const bool reverse,
-	               const Motor_Units encoder_units);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const Motor_Gears gearset, const bool reverse,
+	                     const Motor_Units encoder_units);
 
-	explicit Motor(const std::uint8_t port, const Motor_Gears gearset, const bool reverse);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const pros::Color gearset_color, const bool reverse,
+	                     const Motor_Units encoder_units);
 
-	explicit Motor(const std::uint8_t port, const pros::Color gearset_color, const bool reverse);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const Motor_Gears gearset, const bool reverse);
 
-	explicit Motor(const std::uint8_t port, const Motor_Gears gearset);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const pros::Color gearset_color, const bool reverse);
 
-	explicit Motor(const std::uint8_t port, const pros::Color gearset_color);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const Motor_Gears gearset);
 
-	explicit Motor(const std::uint8_t port, const bool reverse);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const pros::Color gearset_color);
 
-	explicit Motor(const std::uint8_t port);
+	explicit Motor_Group(std::vector<std::int8_t> ports, const bool reverse);
+
+	explicit Motor_Group(std::vector<std::int8_t> ports);
 
```

Comment:
> I also think that accepting `pros::Color` instead of `Motor_Gears` is wholly unnecessary at best and very unclear at worst. Accepting only a couple members of an enum (I assume that's whats happening here?) feels like terrible practice to me.

This was a community request that I don't think is best practice either, but with respect to those who want it it's not hard to support either.

Reply:
> Surely this long list of constructors could be greatly reduced to 3 or 4 using default parameters instead of all these overloads?

I think that using optional arguments would be good to shorten the number of constructors; however, motors do have defaults for many things without us calling VexOS. Making a bunch of the arguments optional would require us to have some redundant/inconsistent default values, and/or more calls to the VexOS. I think it's fine how it is.

Reply:
> Making a bunch of the arguments optional would require us to have some redundant/inconsistent default values, and/or more calls to the VexOS

I don't think this is actually a big deal, but perhaps 
```cpp
explicit Motor_Group(std::initializer_list<std::int8_t> ports, 
                     const std::optional<Motor_Gears> gearset = std::nullopt,
                     const std::optional<bool> reverse = std::nullopt,
                     const std::optional<Motor_Units> encoder_units = std::nullopt);
```
if you're really concerned about it? you would just call that as 
```cpp
Motor_Group({8,1,2}, gearset = Motor_Gears::Blue, reverse = false);
```
so users dont actually need to worry about the `std::optional`. I guess leaving it as is could be fine too though.

Reply:
This would be a nice solution if we decide to change it. I will think about this.